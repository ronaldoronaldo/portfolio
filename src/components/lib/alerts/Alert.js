import React from 'react'
import {
  AlertOverlayStyle,
  AlertContainerStyle,
  AlertTitleStyle,
  AlertMessageStyle,
  ButtonContainer,
} from './Alert.style'
import { Button } from '../buttons'
import PropTypes from 'prop-types'

const Alert = ({ title, message, textButton, onClick, ...rest }) => {
  return (
    <AlertOverlayStyle {...rest} >
      <AlertContainerStyle>
        <AlertTitleStyle>{title}</AlertTitleStyle>
        <AlertMessageStyle>{message}</AlertMessageStyle>
        <ButtonContainer>
          <Button text={textButton} onClick={onClick} />
        </ButtonContainer>
      </AlertContainerStyle>
    </AlertOverlayStyle>

  )
}

Alert.propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  textButton: PropTypes.string.isRequired,
  onClick: PropTypes.func
}

export default Alert
