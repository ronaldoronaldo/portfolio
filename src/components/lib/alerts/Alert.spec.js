import React from 'react'
import { shallow } from 'enzyme'
import Alert from './Alert'

describe('<Alert>', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Alert
        title="titulo"
        message="mensagem"
        onClick={() => console.log('Ok')}
        textButton="ok"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
