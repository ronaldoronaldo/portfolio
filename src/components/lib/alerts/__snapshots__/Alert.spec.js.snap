// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert> should render correctly 1`] = `ShallowWrapper {}`;
